HW/12_09/DZ_12_09_TSK02/DZ_12_09_TSK02/Context.cs

```diff
@@ -0,0 +1,94 @@
+using DZ_12_09_TSK02.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace DZ_12_09_TSK02
+{
+    public class Context : DbContext
+    {
+        public Context(DbContextOptions<Context> options) : base(options)
+        {
+            
+        }
+
+        public DbSet<Book> Books { get; set; }
+        public DbSet<Author> Authors { get; set; }
+        public DbSet<User> Users { get; set; }
+        public DbSet<UserBook> UserBooks { get; set; }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            base.OnModelCreating(modelBuilder);
+
+            modelBuilder.Entity<Book>()
+                .HasKey(b => b.BookId);
+            modelBuilder.Entity<Book>()
+                .Property(b => b.Name)
+                .HasMaxLength(100)
+                .IsRequired();
+
+            modelBuilder.Entity<Book>()
+                .HasOne(b => b.Author)
+                .WithMany(a => a.Books)
+                .HasForeignKey(b => b.AuthorId);
+
+            modelBuilder.Entity<Author>()
+                .HasKey(a => a.AuthorId);
+            modelBuilder.Entity<Author>()
+                .Property(a => a.FirstName)
+                .HasMaxLength(50)
+                .IsRequired();
+            modelBuilder.Entity<Author>()
+                .Property(a => a.FirstName)
+                .HasMaxLength(50)
+                .IsRequired();
+
+            modelBuilder.Entity<User>()
+                .HasKey(u => u.UserId);
+            modelBuilder.Entity<User>()
+                .HasIndex(u => u.Email)
+                .IsUnique();
+            modelBuilder.Entity<User>()
+                .Property(u => u.Email)
+                .HasMaxLength(40);
+
+            modelBuilder.Entity<UserBook>()
+                .HasKey(us => us.UserBookId);
+
+            modelBuilder.Entity<UserBook>()
+                .HasOne(us => us.Book)
+                .WithMany(b => b.UserBooks)
+                .HasForeignKey(us => us.BookId)
+                .OnDelete(DeleteBehavior.Cascade);
+
+            modelBuilder.Entity<UserBook>()
+                .HasOne(us => us.User)
+                .WithMany(b => b.UserBooks)
+                .HasForeignKey(us => us.UserId)
+                .OnDelete(DeleteBehavior.Cascade);
+        }
+
+        protected void SeedingNewData(ModelBuilder modelBuilder)
+        {
+            var authors = Author.Generate(5);
+            var users = User.Generate(5);
+            var books = Book.Generate(5, authors.Select(x => x.AuthorId).ToArray()).ToArray();
+
+            var userbooks = UserBook.Generate(5, 
+                users.Select(x => x.UserId).ToArray(), 
+                books.Select(x => x.BookId).ToArray()).ToArray();
+            
+            modelBuilder.Entity<Author>().HasData(authors);
+            modelBuilder.Entity<User>().HasData(users);
+            modelBuilder.Entity<Book>().HasData(books);
+            modelBuilder.Entity<UserBook>().HasData(userbooks);
+        } 
```

Comment:
It's better not to use Faker for data seeding as we discovered by my example =)
The reason is that it will generate new values every time and new data will be added to migrations.
So if you want to have seeded data you should use constant values, not random